bulldozer/merge.go

```diff
@@ -195,3 +197,30 @@ func allCommits(ctx context.Context, pullCtx pull.Context, client *github.Client
 
 	return repositoryCommits, nil
 }
+
+func postCommentIfUserActionable(ctx context.Context, client *github.Client, owner, repo string, prNum int, githubError github.ErrorResponse) {
+	commentBody := "Error occurred when attempting merge: " + githubError.Message
+
+	switch {
+	case strings.HasPrefix(githubError.Message, "You're not authorized to push to this branch."):
```

Comment:
Since there are already two cases, it would be nice to provide a bit of abstraction around matching these. Something like this?

```go
type ErrorMatcher struct {
  Pattern regexp.Regexp
  Reason  string
}

type (m *ErrorMatcher) Matches(err github.ErrorResponse) bool {
  // ...
}

var UserErrors = []*ErrorMatcher{
  // ...
}
```